expertsystem/ui/_system_control.py

```diff
@@ -465,14 +460,14 @@ def match_external_edges(graphs: List[StateTransitionGraph]) -> None:
     if not graphs:
         return
     ref_graph_id = 0
-    _match_external_edge_ids(graphs, ref_graph_id, get_final_state_edges)
-    _match_external_edge_ids(graphs, ref_graph_id, get_initial_state_edges)
+    _match_external_edge_ids(graphs, ref_graph_id, "get_final_state_edges")
+    _match_external_edge_ids(graphs, ref_graph_id, "get_initial_state_edges")
```

Comment:
Those string changes are super confusing. I assume that change was made in order to handle both normal function and instance methods of the graph. This would require some improvements

Reply:
You're right, this remains the major problem with embedding the functions as methods: you cannot pass around a method as a function argument. Afaics there's no way around this in Python.

So to summarise: having functions like `get_final_state_edges` as a member of `StateTransitionGraph` is safer and easier in usage (keep responsibilities nearby each other), but (_within the current framework!_) the price is this need for `getattr`. 
I hate that too, because these strings are against static typing and things like tab completion.

Reply:
> I assume that change was made in order to handle both normal function and instance methods of the graph.

The thing is that I didn't want to touch the rest of the framework other than `import` statements. In this case, there are a few functions under `ui` that could become part of the `graph` module (take this `_match_external_edge_ids` for example), but that requires additional refactoring that goes beyond what this PR intended.

Reply:
The problem is that, this rather complicated change (which started of with a simple move functions to method idea) was not discussed and planned in prior. I have to at least think about this now, and see if there is some quick and nice solution for this.

Reply:
That's true. Although on the other hand, we probably would have concluded that issue with "sure, sounds good, let's convert the functions to methods", because the consequence of that only pop up after the refactoring itself (this PR). In that sense, issues and PRs cannot be strictly separated.

So I decoupled this PR from #230 (no rebasing) and #143, then we can consider this PR a proposal that can incite other discussions (such as https://github.com/ComPWA/expertsystem/pull/229#discussion_r475536307).

Reply:
Ok, so checked out if there is a nice way with limited amount of changes to solve this. Unfortunately the amount of changes would go indeed beyond the scope of the PR. So we tackle this in another one